test/lean/reg.expected.lean

```diff
@@ -0,0 +1,16 @@
+import Out.Sail.Sail
+
+open Sail
+
+class MonadReg where
```

Comment:
Shouldn't the register state be part of the state that's carried by `SailM`? That would give us the getter and setter "for free" instead? Or do we want to keep this flexible? What's your opinion on this, @bacam ?